libraries/Adapters/Microsoft.Bot.Builder.Adapters.WeChat/Helpers/DictionarySort.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+
+namespace Microsoft.Bot.Builder.Adapters.WeChat
+{
+    public class DictionarySort : IComparer
+    {
+        public int Compare(object oLeft, object oRight)
+        {
+            var leftString = oLeft as string;
```

Comment:
These casts are not valid. Nor are they being checked. 

Please use proper Generic Dictionarry here. This code seems to already existin the .Net framework.